Stripe/STPAddressViewModel.m

```diff
@@ -167,6 +170,60 @@ - (void)addressFieldTableViewCellDidReturn:(STPAddressFieldTableViewCell *)cell
     [[self cellAfterCell:cell] becomeFirstResponder];
 }
 
+- (void)addressFieldTableViewCellDidEndEditing:(STPAddressFieldTableViewCell *)cell {
+    if (cell.type == STPAddressFieldTypeZip) {
+        [self updateCityAndStateFromZipCodeCell:cell];
+    }
+}
+
+- (void)updateCityAndStateFromZipCodeCell:(STPAddressFieldTableViewCell *)zipCell {
+
+    if (!zipCell.textField.validText
+        || ![_addressFieldTableViewCountryCode isEqualToString:@"US"]) {
+        return;
+    }
+
+    NSString *zipCode = zipCell.contents;
+    STPAddressFieldTableViewCell *cityCell = nil;
+    STPAddressFieldTableViewCell *stateCell = nil;
+    for (STPAddressFieldTableViewCell *cell in self.addressCells) {
+        if (cell.type == STPAddressFieldTypeCity) {
+            cityCell = cell;
+        }
+        else if (cell.type == STPAddressFieldTypeState) {
+            stateCell = cell;
+        }
+    }
+
+    if ((cityCell == nil && stateCell == nil)
+        || (cityCell.contents.length > 0 || stateCell.contents.length > 0)) {
+        // Don't auto fill if either have text already
+        // Or if neither are non-nil
+        return;
+    }
+    else {
+
+        CLGeocoder *geocoder = [CLGeocoder new];
+        NSString *zipKey = (NSString *) kABPersonAddressZIPKey;
+        NSString *countryCodeKey = (NSString *) kABPersonAddressCountryCodeKey;
+
+        if (zipKey && countryCodeKey) {
```

Comment:
This should be caught above i think? The first check in the method is that country code is US and zipCode is valid (although maybe nil zip codes are valid? I can check)